pubsub/rocketmq/metadata.go

```diff
@@ -14,63 +14,157 @@ limitations under the License.
+	ProducerGroupName string `mapstructure:"producerGroupName"`
+	NameSpace         string `mapstructure:"nameSpace"`
+	NameServerDomain  string `mapstructure:"nameServerDomain"`
+	NameServer        string `mapstructure:"nameServer"`
+	AccessKey         string `mapstructure:"accessKey"`
+	SecretKey         string `mapstructure:"secretKey"`
+	SecurityToken     string `mapstructure:"securityToken"`
+	Retries           int    `mapstructure:"retries"`
 
-func getDefaultRocketMQMetaData() *rocketMQMetaData {
-	return &rocketMQMetaData{
-		AccessProto:       "",
-		AccessKey:         "",
-		SecretKey:         "",
-		NameServer:        "",
-		GroupName:         "",
-		NameSpace:         "",
-		ConsumerGroup:     "",
-		ConsumerBatchSize: 0,
-		NameServerDomain:  "",
-		ContentType:       pubsub.DefaultCloudEventDataContentType,
-		Retries:           3,
-		SendTimeOut:       10,
-	}
+	// Producer Queue selector
+	// There are five implementations of queue selector，Hash、Random、Manual、RoundRobin、Dapr，respectively
+	//
+	// Dapr Queue selector is design by dapr developers
+	ProducerQueueSelector QueueSelectorType `mapstructure:"producerQueueSelector"`
+
+	// Message model defines the way how messages are delivered to each consumer clients
+	// 	RocketMQ supports two message models: clustering and broadcasting. If clustering is set, consumer clients with
+	// 	the same {@link #ConsumerGroup} would only consume shards of the messages subscribed, which achieves load
+	// 	balances; Conversely, if the broadcasting is set, each consumer client will consume all subscribed messages
+	// 	separately.
+	//
+	// This field defaults to clustering.
+	ConsumerModel string `mapstructure:"consumerModel"`
+
+	// Consuming point on consumer booting.
+	// There are three consuming points:
+	//   - CONSUME_FROM_LAST_OFFSET: consumer clients pick up where it stopped previously. If it were a newly booting up
+	//   consumer client, according aging of the consumer group, there are two cases.
+	//     cases1:
+	//       if the consumer group is created so recently that the earliest message being subscribed has yet
+	//       expired, which means the consumer group represents a lately launched business, consuming will
+	//       start from the very beginning.
+	//     case2:
+	//       if the earliest message being subscribed has expired, consuming will start from the latest messages,
+	//       meaning messages born prior to the booting timestamp would be ignored.
+	//   - CONSUME_FROM_FIRST_OFFSET: Consumer client will start from earliest messages available.
+	//   - CONSUME_FROM_TIMESTAMP: Consumer client will start from specified timestamp, which means messages born
+	//   prior to {@link #consumeTimestamp} will be ignored
+	FromWhere string `mapstructure:"fromWhere"`
+
+	/**
+	 * Backtracking consumption time with second precision. Time format is
+	 * 20131223171201<br>
+	 * Implying Seventeen twelve and 01 seconds on December 23, 2013 year<br>
```

Comment:
I am not understanding this comment

Reply:
The comment is that I copy the RocketMQ comment.
ConsumeTimestamp indicates that the consumer can start consuming from the current point in time in the format of 20131223171201(yyyyMMddHHmmss).